### ISSUE TITLE
[Bug]: pyxis SPANK plugin not loaded on worker nodes due to Slurm configless mode incompatibility

### ISSUE THREAD

**Issue by DTM-7-RX** [aws-samples/sample-physical-ai-scaffolding-kit#5]
## Prerequisites

- [x] I have searched existing issues to make sure this bug has not already been reported.
- [x] I have verified the issue against the latest `main` branch.

## Environment

- AWS Service: SageMaker HyperPod
- Instance type: ml.g5.xlarge (worker), ml.c5.large (controller/login)
- Number of worker nodes: 1
- OS / AMI: HyperPod default AMI (Ubuntu 22.04)
- Training framework: NVIDIA Isaac GR00T N1.6 (PyTorch-based)
- CUDA / Driver version: CUDA 12.x (from nvcr.io/nvidia/pytorch:25.04-py3)
- Container image: gr00t-train:latest (built via samples/gr00t/training/Dockerfile)
- Scheduler: Slurm (configless mode, as deployed by HyperPod)

## Bug Description

**Expected behavior:**

After CDK deployment, `#SBATCH --container-image=...` and `srun --container-image=...` in Slurm job scripts should launch jobs inside the specified Enroot container. The pyxis SPANK plugin should be properly registered with slurmd on worker nodes.

**Actual behavior:**

The `--container-image` directive is silently ignored. Jobs run directly on the host OS instead of inside the container. This causes failures because container-only paths (e.g., `/workspace/gr00t`) do not exist on the host.

This affects **all PASK samples that use Enroot containers**, including both GR00T (`#SBATCH --container-image`) and OpenPI (`srun --container-image`).

**Root cause:**

HyperPod runs Slurm in **configless mode**, where worker nodes pull configuration from the controller (slurmctld) at startup. The Slurm documentation explicitly states:

> "If any of the supported config files 'Include' additional config files, the Included configs will NOT be shipped to the slurmds."
> — https://slurm.schedmd.com/configless_slurm.html

The lifecycle script (`install_enroot_pyxis.sh` from [awsome-distributed-training](https://github.com/aws-samples/awsome-distributed-training)) registers pyxis by adding an `include` directive to `/opt/slurm/etc/plugstack.conf`:

```
required /opt/slurm/lib/slurm/spank_auto_resume.so
include /opt/slurm/etc/plugstack.conf.d/pyxis.conf
```

In configless mode, this `include` line and the referenced `pyxis.conf` are never delivered to the worker node's conf-cache (`/var/spool/slurmd/conf-cache/plugstack.conf`). As a result, slurmd on the worker node is unaware of the pyxis plugin.

**Evidence from a clean deployment:**

Controller node (`/opt/slurm/etc/plugstack.conf`):
```
required /opt/slurm/lib/slurm/spank_auto_resume.so
include /opt/slurm/etc/plugstack.conf.d/pyxis.conf    ← present
```

Worker node (`/var/spool/slurmd/conf-cache/plugstack.conf`):
```
required /opt/slurm/lib/slurm/spank_auto_resume.so
                                                       ← include line missing
```

This issue is **not instance-type dependent** — it affects all HyperPod environments using configless mode with these lifecycle scripts.

## Steps to Reproduce

1. Deploy a fresh HyperPod cluster using PASK CDK:
   ```bash
   cd hyperpod && npm install
   AWS_DEFAULT_REGION=us-east-1 npx cdk deploy --require-approval never
   ```

2. Follow the GR00T training guide (Phase 1–3) to clone repos, build the Docker image, and import the Enroot container.

3. Submit a fine-tuning job:
   ```bash
   cd /fsx/ubuntu/sample-physical-ai-scaffolding-kit/samples/gr00t/training
   sbatch slurm_finetune_container.sh
   ```

4. Observe the job fails immediately with:
   ```
   cd: /workspace/gr00t: No such file or directory
   ```

5. Verify the conf-cache on the worker node:
   ```bash
   cat /var/spool/slurmd/conf-cache/plugstack.conf
   # Only contains spank_auto_resume.so, no pyxis reference
   ```

6. Apply the workaround on the worker node:
   ```bash
   sudo bash -c 'echo "include /opt/slurm/etc/plugstack.conf.d/pyxis.conf" >> /var/spool/slurmd/conf-cache/plugstack.conf'
   sudo systemctl restart slurmd
   ```

7. Re-submit the same job — it now runs inside the container successfully (model loads, dataset initializes).

## Relevant Log Output

```shell
# Job output (without pyxis fix):
==================================================
GR00T Fine-tuning - Container Job
==================================================
Job ID: 3
GR00T_HOME: /workspace/gr00t
==================================================

# Job error:
/var/spool/slurmd/job00003/slurm_script: line 64: cd: /workspace/gr00t: No such file or directory
```

## Suggested Fix

Instead of relying on `include` in `plugstack.conf`, write the pyxis configuration inline. This avoids the configless mode limitation:

```bash
# In the lifecycle script, replace:
echo "include /opt/slurm/etc/plugstack.conf.d/pyxis.conf" >> /opt/slurm/etc/plugstack.conf

# With:
echo "required /usr/local/lib/slurm/spank_pyxis.so" >> /opt/slurm/etc/plugstack.conf
```

Alternatively, the lifecycle script could update the worker's conf-cache directly after installing pyxis:

```bash
echo "required /usr/local/lib/slurm/spank_pyxis.so" >> /var/spool/slurmd/conf-cache/plugstack.conf
systemctl restart slurmd
```

Note: The upstream lifecycle script is from [awsome-distributed-training](https://github.com/aws-samples/awsome-distributed-training/tree/main/1.architectures/5.sagemaker-hyperpod/LifecycleScripts/base-config). A fix there would benefit all downstream consumers.

**Comment by sparkgene** [comment]
merged fix
